Unreal Engine plugin documentation update (#7626)
getsentry/sentry-docs: src/platform-includes/set-environment/unreal.mdx

```cpp
FConfigureSettingsDelegate SettingsDelegate;
SettingsDelegate.BindDynamic(this, &USomeClass::HandleSettingsDelegate);

void USomeClass::HandleSettingsDelegate(USentrySettings* Settings)
{
    Settings->Environment = FString(TEXT("MyEnvironment"));
}

...

USentrySubsystem* SentrySubsystem = ...;

SentrySubsystem->InitializeWithSettings(SettingsDelegate);
```

> By default, the SDK reports `Editor` when running inside the Unreal Engine Editor. Otherwise, the default environment can be `Release`, `Development` or `Debug` depending on build configuration. Also, it's possible override environment value manually.

Reviewer: ```suggestion
By default, the SDK reports `Editor` when running inside the Unreal Engine Editor. You can also set the default environment to `Release`, `Development`, or `Debug` in your build configuration or override the environment value manually.
```
